### DISCUSSION TITLE
Is it possible to rebuild/refresh Activity in Workflow Designer?

### DISCUSSION THREAD

**Discussion by myushchenko** [elsa-workflows/elsa-core#503] [Q&A]
Hello, 

Now in my solution I have my custom activity with **Select** property and I need it refresh after reload page:
```csharp
        [ActivityProperty(
            Type = ActivityPropertyTypes.Select,
            Hint = "Please select a page"
        )]
        [SelectOptions(nameof(GetPages), typeof(PageInteractionActivity))]
        public string Page
        {
            get => GetState(() => string.Empty, nameof(Page));
            set => SetState(value, nameof(Page));
        }

        public static IEnumerable<SelectOption<object>> GetPages()
        {
            return PageDefinitionHelper.GetPages();
        }
```

PageDefinitionHelper.GetPages() return list of pages from API. But now it run only one time, because it reflect all properties from Activity class during application setup.

Should it possible force it by hand? 
Thanks.

**Comment by sfmskywalker** [comment]
Unfortunately this is not possible right now, but it is certainly a capability I am planning on adding. Not for the Elsa 1 designer, but for the Elsa 2 version. Should you need it for Elsa 1, I'm happy to to accept a PR for that of course.

**Reply by parikhpratik** [reply]
In ELSA2, How we can update custom activities without restarting ELSA engine and also how we can update custom activities in ELSA dashboard without restart?
